monitor_handler.go

```diff
@@ -34,7 +34,8 @@ func (m *monitor) handleMonitor(w http.ResponseWriter, r *http.Request) {
 
 	for _, errstr := range m.sigerrstrs {
 		if errstr != "" {
-			httpError(w, r, http.StatusInternalServerError, errstr)
+			// "%s" is a go vet false positive correction
```

Comment:
Uhh, I think `go vet` is right in this case. It seems like `httpError()` seems to accept format strings, so if `errstr` contains any percent characters it will expect more args on the call stack.

Reply:
I mean, technically I suppose? But the errors are known to not contain format strings (see, for instance, [these lines](https://github.com/mozilla-services/autograph/blob/5172bfb5ebfddc160172926a3d52e1ea4f39bae4/monitor.go#L78-L88)), and if they do, this isn't C: the `fmt.Sprintf` fills them in with `EXTRA` strings. See https://pkg.go.dev/fmt#hdr-Format_errors for how Go handles format string errors

The alternative would be to write another httpError that just didn't take extra args at all and that didn't seem worth it.

Reply:
Like, the violation that pops up is actually because of the `Sprintf` call, but it doesn't go away when I do this work conditionally (because Sprintf doesn't actually care much). The linter is just pattern matching on "string + args in function signature". See this patch and how it doesn't make the lint violation go away:

```patch
diff --git a/errors.go b/errors.go
index 9818b5b1..ba636d0b 100644
--- a/errors.go
+++ b/errors.go
@@ -23,7 +23,13 @@ func httpError(w http.ResponseWriter, r *http.Request, errorCode int, errorMessa
 		"code": errorCode,
 		"rid":  rid,
 	}).Errorf(errorMessage, args...)
-	msg := fmt.Sprintf(errorMessage, args...)
+	var msg string
+	if len(args) == 0 {
+		msg = errorMessage
+	} else {
+		msg = fmt.Sprintf(errorMessage, args...)
+	}
+
 	msg += "\r\nrequest-id: " + rid
 	// when nginx is in front of go, nginx requires that the entire
 	// request body is read before writing a response.
diff --git a/monitor_handler.go b/monitor_handler.go
index da97be8c..745ce17f 100644
--- a/monitor_handler.go
+++ b/monitor_handler.go
@@ -34,8 +34,7 @@ func (m *monitor) handleMonitor(w http.ResponseWriter, r *http.Request) {
 
 	for _, errstr := range m.sigerrstrs {
 		if errstr != "" {
-			// "%s" is a go vet false positive correction
-			httpError(w, r, http.StatusInternalServerError, "%s", errstr)
+			httpError(w, r, http.StatusInternalServerError, errstr)
 			return
 		}
 	}
```

Reply:
I guess maybe you're saying you just want the comment deleted?

Reply:
Yeah, that's all I was trying to say - just remove the comment and it's fine.

Reply:
*embarrassed face* can do